pkg/statedb/types.go

```diff
@@ -56,6 +56,19 @@ type Table[Obj any] interface {
 	// when anything in the table changes as more fine-grained notifications
 	// are not possible with a lower bound search.
 	LowerBound(ReadTxn, Query[Obj]) (iter Iterator[Obj], watch <-chan struct{})
+
+	// DeleteTracker creates a new delete tracker for the table
+	// starting from the given revision.
+	//
+	// WARNING: Do not create a WriteTxn against this table and then call this
+	// with a ReadTxn. This will lead to a dead lock as in order for DeleteTracker()
+	// to be usable by readers it needs to perform a write transaction on behalf
+	// of the caller.
+	//
+	// If a suitable WriteTxn is provided, then that is used to add the tracker.
+	// In that case the tracker will only become active when the transaction is
+	// committed.
+	DeleteTracker(txn ReadTxn, trackerName string) (*DeleteTracker[Obj], error)
```

Comment:
Or we just keep it as `WriteTxn` and then `DeleteTracker` is the only usable method that someone without `RWTable` can use to "modify the table", e.g. readers who have `Table[Obj]` can do `WriteTxn(table)` and call `DeleteTracker` with it. Would get rid of this deadlock potential.

I think it's a reasonable trade-off to still expose `WriteTxn` against even read-only tables for everyone as it allows this DeleteTracker() stuff and it allows building smart writers on top of `RWTable[Obj]` while still allowing cross-table transactionality. The downside I see is that then arbitrary code can "lock" the table even when they can't modify it otherwise. Is this practical engineering or am I being reckless?